### ISSUE TITLE
Inconsistent result from write queries chained after getAll

### ISSUE THREAD

**Issue by simonratner** [neumino/thinky#181]
For a singular result, write query resolves to a single model; for multiple results it resolves to an array:
```javascript
Model.getAll(1, {index: "secondary"}).update({...}).run().then(function(docs) {
  docs.forEach(...); /// docs may be an array, or it may not, in which case :( 
});
```

The culprit is https://github.com/neumino/thinky/blob/master/lib/query.js#L140, which I am guessing exists to support the Model.get().update() use case. Really need a better way to distinguish singular queries from plural ones.

**Comment by neumino** [comment]
Ah that's a tricky one.

In the case of `update`, just looking at the result of the query, you cannot know if it comes from
```
r.table('...').get(...).update({...})
// or
r.table('...').getAll(...).update({...})
```

I guess we'll have to look at the query itself. If the last term is `get`, we pass an object, if it's not we give back an array.
It's not pretty pretty, but we already do that in a few places.

Thanks for the report @simonratner and looking in the code!

**Comment by neumino** [comment]
Fixed in 2.0.1
